client/packages/system/src/Immunisation/ProgramDetailView/Toolbar/Toolbar.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { FC, PropsWithChildren } from 'react';
+import {
+  AppBarContentPortal,
+  BasicTextInput,
+  Grid,
+  useTranslation,
+} from '@openmsupply-client/common';
+import { DraftImmunisationProgram } from '../../api/hooks/useImmunisationProgram';
+
+interface ToolbarProps {
+  onUpdate: (patch: Partial<DraftImmunisationProgram>) => void;
+  draft: DraftImmunisationProgram;
+  isError: boolean;
+  error: string;
+}
+
+export const Toolbar: FC<PropsWithChildren<ToolbarProps>> = ({
+  onUpdate,
+  draft,
+  isError,
+  error,
+}: ToolbarProps) => {
+  const t = useTranslation('system');
+  return (
+    <AppBarContentPortal sx={{ display: 'flex', flex: 1, marginBottom: 1 }}>
+      <Grid container>
+        <Grid item display="flex" flex={1} flexDirection="column" gap={1}>
+          <BasicTextInput
+            fullWidth
+            value={draft.name}
+            onChange={e =>
```

Comment:
I wonder if onBlur better here with dirty check for saving?

I also wonder if a different UI would be better so the name seems more like 'info' about the page the user is on rather than an editable field